server/src/db/catalog/chunk.rs

```diff
@@ -732,6 +732,16 @@ impl CatalogChunk {
         }
     }
 
+    /// Return parquet chunk if it is persisted
+    pub fn parquet_chunk(&self) -> Option<Arc<ParquetChunk>> {
```

Comment:
```suggestion
    pub fn parquet_chunk(&self) -> Option<&Arc<ParquetChunk>> {
```

Pet peeve of mine, but where there is no reason to do so, it is better to return & vs a cloned reference - this allows borrow chaining in e.g. map to work correctly, otherwise it complains about a dropped temporary.

(there is also the minor point of avoiding unnecessary atomic operations)